zookeeper-server/src/test/java/org/apache/zookeeper/test/DisconnectedWatcherTest.java

```diff
@@ -221,6 +228,7 @@ public void testManyChildWatchersAutoReset() throws Exception {
         watcher.waitForDisconnected(30000);
         startServer();
         watcher.waitForConnected(30000);
+        watcher1.waitForConnected(30000);
```

Comment:
If this is fixing this test - that's great.
Trying to understand why. If zk1 isn't connected, should we get CONNECTIONLOSS on line 237 zk1.create?
Here is an example and I don't see it.
https://builds.apache.org/job/ZooKeeper_branch35_java10/263/testReport/junit/org.apache.zookeeper.test/DisconnectedWatcherTest/testManyChildWatchersAutoReset/

Good improvement anyway, I think should be merged.

Reply:
That example is due to an unrelated class of failure, in which the Zookeeper server is permanently unreachable (notice that the entire test timed out); I think many failure cases of that kind are the product of failure to bind() the dynamically-assigned port to the server socket (probably due to collisions with other tests running concurrently on the same build host).  I've updated the description of this PR to explain what kind of failure it's intended to fix.